test/GoodGhosting.test.js

```diff
@@ -367,6 +376,55 @@ contract("GoodGhosting", (accounts) => {
         });
     });
 
+
+    describe("when a player early withdraws in segment 0", async () => {
+        it("reverts if the same user tries to rejoin the game in a latter segment", async() => {
+            await approveDaiToContract(player1);
+            await goodGhosting.joinGame(whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof, { from: player1 });
+            await timeMachine.advanceTime(weekInSecs);
+            await goodGhosting.earlyWithdraw({ from: player1 });
+            await approveDaiToContract(player1);
+            await truffleAssert.reverts(goodGhosting.joinGame(whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof, { from: player1 }), "Game has already started");
+        });
+
+        it("reverts if a user tries to rejoin the game in segment 0 without doing an early withdraw", async() => {
+            await approveDaiToContract(player1);
+            await goodGhosting.joinGame(whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof, { from: player1 });
+            await approveDaiToContract(player1);
+            await truffleAssert.reverts(goodGhosting.joinGame(whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof, { from: player1 }), "Cannot join the game more than once");
+        });
+
+        it("user can rejoin the game on segment 0 after an early withdrawal", async() => {
+            await approveDaiToContract(player1);
+            const playerAllowance = await token.allowance(player1, goodGhosting.address);
+            assert(playerAllowance.gte(segmentPayment));
+            await goodGhosting.joinGame(whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof, { from: player1 });
+            await goodGhosting.earlyWithdraw({ from: player1 });
+            await approveDaiToContract(player1);
+            await goodGhosting.joinGame(whitelistedPlayerConfig[0][player1].index, whitelistedPlayerConfig[0][player1].proof, { from: player1 });
+            const playerInfo = await goodGhosting.players(player1);
+            assert(playerInfo.mostRecentSegmentPaid.eq(new BN(0)));
+            assert(playerInfo.amountPaid.eq(segmentPayment));
+            assert(playerInfo.canRejoin ===  false);
+            assert(playerInfo.withdrawn ===  false);
```

Comment:
imho, for clarity, I'd prefer to have this as a specific test case, where we explicitly state what we're testing. let me know your thoughts.